Runtime/Scripts/LyumaAv3Emulator.cs

```diff
@@ -187,102 +185,126 @@ private void Start()
 			Camera.onPreCull += PreCull;
 			Camera.onPostRender += PostRender;
 			emulatorInstance = this;
-			RunPreprocessors();
 			if (WorkaroundPlayModeScriptCompile) {
 				LyumaAv3Runtime.ApplyOnEnableWorkaroundDelegate();
 			}
-
-			InitNextFrame = true;
+			AddMarkersToDisabled();
+			InitializeInScene();
 		}
 
-		private void RunPreprocessors()
+		[DefaultExecutionOrder(-9999)]
+		private class LyumaAv3Marker : MonoBehaviour
 		{
-			List<GameObject> avatars = AvatarList;
-			foreach (var avadesc in avatars)
+			private void Start()
 			{
-				bool alreadyHadComponent = avadesc.gameObject.GetComponent<LyumaAv3Runtime>() != null;
-				if (RunPreprocessAvatarHook && !alreadyHadComponent)
+				if (GetComponent<VRCAvatarDescriptor>() != null)
 				{
-					GameObject origClone = GameObject.Instantiate(avadesc.gameObject);
-					origClone.name = avadesc.gameObject.name;
-					avadesc.gameObject.name = origClone.name + "(Clone)";
-					LyumaAv3Runtime.InvokeOnPreProcessAvatar(avadesc.gameObject);
-					avadesc.gameObject.name = origClone.name;
-					GameObject.DestroyImmediate(origClone);
+					Debug.Log("Setting up Emulator late on: " + gameObject.name);
+					emulatorInstance.RunPreprocessors(gameObject);
+					emulatorInstance.Initialize(gameObject);
 				}
 			}
 		}
 		
-		private void Initialize()
+		private void AddMarkersToDisabled()
+		{
+			Resources.FindObjectsOfTypeAll<VRCAvatarDescriptor>()
+				.Select(x => x.gameObject)
+				.Where(x => !x.activeSelf)
+				.ToList()
+				.ForEach(x => x.AddComponent<LyumaAv3Marker>());
+		}
+
+		private void RunPreprocessors(GameObject avatar)
+		{
+			bool alreadyHadComponent = avatar.gameObject.GetComponent<LyumaAv3Runtime>() != null;
+			if (RunPreprocessAvatarHook && !alreadyHadComponent)
+			{
+				GameObject origClone = GameObject.Instantiate(avatar.gameObject);
+				origClone.name = avatar.gameObject.name;
+				avatar.gameObject.name = origClone.name + "(Clone)";
+				LyumaAv3Runtime.InvokeOnPreProcessAvatar(avatar.gameObject);
+				avatar.gameObject.name = origClone.name;
+				GameObject.DestroyImmediate(origClone);
+			}
+		}
+
+		private void InitializeInScene()
 		{
-			List<GameObject> avatars = AvatarList;
+			List<GameObject> avatars = GameObject.FindObjectsOfType<VRCAvatarDescriptor>()
```

Comment:
since FindObjectsOfType is static method of `Object`, I think call like `Object.FindObjectsOfType()`. to avoid conflict with Syatem.Object, we can use `using Object = UnityEngine.Object` at the top.